data/presets/presets/amenity/bicycle_parking/building.json

```diff
@@ -0,0 +1,25 @@
+{
+    "icon": "maki-bicycle",
+    "fields": [
+        "{amenity/bicycle_parking}",
+        "opening_hours"
+    ],
+    "geometry": [
+        "point",
+        "vertex",
+        "area"
+    ],
+    "tags": {
+        "amenity": "bicycle_parking",
+        "bicycle_parking": "building"
```

Comment:
hmm this one is weird..
Is it really `bicycle_parking=building` and not `building=bicycle_parking`?
Can we change that before it gains traction?

Reply:
These are two different things:

`bicycle_parking=building`  defines the type of bicycle parking: That it is in an enclosed building.

`building=bicycle_parking` is not defined in the wiki, but I guess it would be a building that was built for the purpose of serving as a bicycle parking station.

The difference is, that the bicycle parking doesn't have to fill the whole building but may be some area or point within the building. So, much like `parking=multi-storey` vs `building=parking`.

The `bicycle_parking=building` is already used by StreetComplete by a year or so. And much longer documented like that.

Reply:
I agree with Bryan, this preset is confusing. I'd prefer to just use `indoor=yes` and reserve the `bicycle_parking` tag for the actual method, like stands, lockers, or whatnot.

Reply:
`indoor` is for indoor-mapping only but `bicycle_parking=building` may and is often tagged on the whole building (but doesn't have to, as mentioned).

Reply:
On the wiki page, someone else raised a similar concern as Bryan in 2017: https://wiki.openstreetmap.org/wiki/Talk:Key:bicycle_parking#Not_sure_if_bicycle_parking.3Dbuilding_makes_sense

Reply:
To summarize, tagging [bicycle parking stations](https://en.wikipedia.org/wiki/Bicycle_parking_station) as `bicycle_parking=building` is missing an alternative because it could be tagged as an indoor-node  but also the building itself. I see three options, please choose:
1. drop the  bicycle parking station preset from this PR
2. limit the bicycle parking station preset to only areas and set `building=bicycle_parking` instead
3. use `bicycle_parking=building` after all, leave PR as-is

Reply:
Hmm I was assuming this preset was for bike parking inside buildings with other purposes, but I guess [bike parking garages are a thing](https://www.psfk.com/2017/08/new-bicycle-garage-netherlands-largest-in-the-world.html)? If that's what this tag is actually for, then I'm fine with option 2 as long as we rename the preset to "Bicycle Parking Garage" or similar.

Reply:
Yes, the Netherlands are known worldwide for their advanced bicycle infrastructure. They call these apparently mostly "[fietsenstalling stationsplein](https://www.bezoek-utrecht.nl/wat-te-doen/fietsen/fietsenstallingen/3440301639/fietsenstalling-stationsplein)" (literally: bicycle storage station square).

[Here is a Google StreetView of such a location](https://www.google.com/maps/@52.0929459,5.1137695,3a,55.7y,143.28h,84.77t/data=!3m6!1e1!3m4!1scRA6JV6ql6h0389fOHS7zg!2e0!7i13312!8i6656) (note that bicycle "street" ). [The same location in OpenStreetMap](https://www.openstreetmap.org/node/2466665935) is already tagged.
Though, note that with option 2, the preset would not work for cases like these because it is mapped as a node, much like a `amenity=parking_entrance`.

I will change the PR to option 2 later this evening (UTC+2) or otherwise on Saturday.